blog/2023-03-31-remote-unix-monitoring-with-net-snmp/index.md

```diff
@@ -0,0 +1,407 @@
+
+The general configuration for a disk space monitoring chart looks like:
+
+```yaml
+  - family: disk_space
+    id: _
+    title: /
+    type: stacked
+    units: bytes
+    dimensions:
+    - {algorithm: absolute, name: avail, oid: 1.3.6.1.4.1.2021.9.1.7.1}
+    - {algorithm: absolute, name: used, oid: 1.3.6.1.4.1.2021.9.1.8.1}
+```
+
+The above configuration snippet can be used to monitor the root filesystem’s space usage on any system. To get
+a list of what other filesystems you can monitor, run:
+
+```sh
+snmpwalk -v 2c -c netdata HOSTNAME 1.3.6.1.4.1.2021.9.1.2
+```
+
+or, if using SNMP v3:
+
+```sh
+snmpwalk -v 3 -u netdata -l noAuthPriv -a SHA -A PASSWORD HOSTNAME 1.3.6.1.4.1.2021.9.1.2
+```
+
+Those should produce output that looks something like:
+
+```
+UCD-SNMP-MIB::dskPath.1 = STRING: /
+UCD-SNMP-MIB::dskPath.2 = STRING: /dev
+UCD-SNMP-MIB::dskPath.3 = STRING: /boot/efi
+```
+
+The relevant parts here are the number just before the `=`, which is the disk index, and the part at the end of
+the line, which indicates the path to the filesystem corresponding to that index. You can replace the final `1`
+in the OIDs in the configuration above with the desired disk index to monitor that specific disk, thoughy you
+should also change the `id` and `title` for the chart to uniquely identify that specific disk.
+
+Index 1 will always be the root filesystem. The indexes of other filesystems are based on the order they are listed
+in the kernel’s mount table, so they are dependent on the exact configuration of the system, and they may change
+as a result of seemingly unrelated changes such as kernel upgrades.
+
+### Virtual Nodes
+
+Starting with v1.39.0, Netdata is adding a new feature called ‘virtual nodes’ (or ‘vnodes’). Virtual nodes
+let you treat a set of metrics as a separate system from the system that Netdata is running on, which is perfect
+for monitoring remote UNIX systems via SNMP.
+
+You can open the virtual nodes configuration by running:
+
+```sh
+/etc/netdata/edit-config vnodes/vnodes.conf
+```
+
+The configuration file itself is a YAML document, just like with the SNMP collector configuration.
+
+A simple configuration entry for a single virtual node might look like this:
+
+```yaml title="/etc/netdata/vnodes/vnodes.conf"
+- hostname: foo.example.com                  # This defines the hostname that will be shown for the node
+  guid: 00000000-0000-0000-0000-000000000000 # This defines the node’s GUID. It should be unique among all nodes.
+  labels:                                    # This defines host labels, which are used to present information about the node in Netdata Cloud
+    _architecture: x86_64
+    _os_name: Solaris
+    _os_version: 11
+    _system_cores: 8
+    _system_cpu_freq: 4700000000
+    _system_disk_space: 1099511627776
+    _system_ram_total: 34359738368
+    _virtualization: kvm
+```
+
+Once you have a virtual node defined, you simply need to add a `vnode:` key to the SNMP job that you want to
```

Comment:
@Ferroin wondering if/can or should we have some sort of image to show how the vnode `foo.example.com` ends up looking here in Netdata Cloud?

Reply:
I’m not sure if it’s worth it? Virtual nodes look no different from any other node, they just have a different set of collectors listed.